### DISCUSSION TITLE
random generate non-overlapping edge network

### DISCUSSION THREAD

**Discussion by tliu1023** [networkx/networkx#5412] [Q&A] [answered]
Hi,

I'm wondering is there a way to randomly generate the network where the edges are not overlapping? Like a road map network. Thank you!

Best,
Tong

**Comment by dschult** [comment, marked as answer]
The standard graph has no concept of "overlapping", so we don't have that currently in the package. It sounds like you would need 2-D positions for the nodes and paths for the edges to tell if they overlap or not. This might be related to "planar graphs" which can be drawn in a plane without edges crossing.  So, I can imagine some sort of trial-and-error style edge generation routine that checks for whether the graph is still planar and rejects the edge if it isn't.  This would probably be slow.

If you are up for it, try looking at the code for `gnp_random_graph` (networkx/generators/random_graphs.py) and in the main loop where edges are added, put a check for whether the graph is still a planar graph: `check_planarity`.

**Reply by tliu1023** [reply]
Thank you! I will check the code and planar graph
